server.py

```diff
@@ -36,17 +36,25 @@ def index():
 ######################################################################
 @app.route('/products', methods=['GET'])
 def list_products():
-    category        = request.args.get('category')
-    discontinued    = str2bool(request.args.get('discontinued'))
-    results     = []
-    for key, value in products.iteritems():
-        if matchesClause(category,      value['category']) and \
-           matchesClause(discontinued,  value['discontinued']):
-             results.append(products[key])
-    return reply(results, HTTP_200_OK)
+	category=request.args.get('category')
+	discontinued = str2bool(request.args.get('discontinued'))
+	search_term=(request.args.get('search')).lower()
+	#print ("Category", category)
+	#print ("discontinued", discontinued)
+	#print ("search_term", search_term)
+	results = []
+	for key, value in products.iteritems():
+		if category is None and discontinued is None and search_term is None:
+			results.append(products[key])
+		elif matchesClause(category, value['category']) \
+			and matchesClause(discontinued, value['discontinued']):
+			results.append(products[key])
+		elif search_term in value['category'].lower() or search_term in value['name'].lower():
```

Comment:
The name of the https://github.com/NYU-DevOps-Charlie-CSCI-GA-3033-013/products/issues/18 is 'Search key by description and name'. Please add a search for description.

I would opine that the main value of the search feature is the ability to search the 'description' attribute as it is more verbose and offers addition information.